gadget-io/Cargo.toml

```diff
@@ -9,52 +9,51 @@ homepage.workspace = true
 
 [dependencies]
 cfg-if = { workspace = true }
-tracing = { workspace = true, default-features = false, features = ["log", "attributes"] }
 hex = { workspace = true }
-scale-info = { workspace = true}
-parity-scale-codec = { workspace = true }
+multiaddr = { workspace = true, default-features = false }
+parity-scale-codec = { workspace = true, optional = true }
+sc-keystore = { workspace = true, optional = true }
+scale-info = { workspace = true, optional = true }
 serde = { workspace = true }
 sp-application-crypto = { workspace = true, default-features = false, features = ["full_crypto"] }
 sp-core = { workspace = true, features = ["serde", "full_crypto"] }
-multiaddr = { workspace = true, default-features = false }
-url = { workspace = true, default-features = false, features = ["alloc", "serde"] }
+sp-keystore = { workspace = true, optional = true }
+structopt = { workspace = true, optional = true }
 thiserror = { workspace = true }
-tsify = { workspace = true, optional = true }
-wasm-bindgen = { workspace = true, optional = true }
 tokio = { workspace = true, optional = true, features = [
     "time",
     "rt",
     "sync",
 ] }
+tracing = { workspace = true, default-features = false, features = ["log", "attributes"] }
+url = { workspace = true, default-features = false, features = ["alloc", "serde"] }
 
-[target.'cfg(target_family = "wasm")'.dependencies]
-wasm-bindgen-futures = { workspace = true }
-
-serde-wasm-bindgen = { workspace = true }
-js-sys = { workspace = true }
-tokio = { workspace = true, features = ["sync", "macros", "io-util", "rt", "time"] }
-wasmtimer = { workspace = true }
-
-p256 = { workspace = true, features = ["alloc", "ecdsa"], default-features = false }
-
-
-[target.'cfg(not(target_family = "wasm"))'.dependencies]
-sp-keystore = { workspace = true }
-sc-keystore = { workspace = true }
-structopt = { workspace = true }
+# WASM
+js-sys = { workspace = true, optional = true }
+serde-wasm-bindgen = { workspace = true, optional = true }
+tsify = { workspace = true, optional = true }
+wasm-bindgen = { workspace = true, optional = true }
+wasm-bindgen-futures = { workspace = true, optional = true }
+wasmtimer = { workspace = true, optional = true }
 
 [features]
 default = ["std"]
 std = [
     "dep:tokio",
     "sp-application-crypto/std",
+    "dep:sp-keystore",
```

Comment:
Why do we need the `dep` prefix? The `sp` dependencies can compile without default features. Other ones that should compile without them are
- scale-info
- parity-scale-codec
- (maybe structopt)?

Reply:
I just moved them in there since they don't get used outside of the standard keystore.